Finished github actions

PrestonH2O opened the pull request:
Code coverage is posted correctly, looks like all is well. Awaiting final change requests.

cwsmith commented on .github/workflows/cmake.yml:
```diff
@@ -13,13 +13,14 @@ jobs:
     strategy:
       matrix:
-        compiler: [g++,clang++]
+        compiler: [g++]
```
Please make a new workflow (`coverage.yml` ?) that just does coverage using this `yml`.  We want to restore the old one that tests with debug and release mode and multiple compilers (even if it is broken).

PrestonH2O replied:
Do you want me to remove the codeql analysis stuff from the coverage.yml file since it isn't needed for coverage as I understand and will be redundant if we have it in the original version

PrestonH2O replied:
Also there is no need to restore the old one as long as we don't have a "cmake.yml" in pw/lcovGithubActions when the pull request is complete it will just have the old one from NM-rework, right?

cwsmith replied:
codeql: Yes, please remove.

yml: Correct.  Yes, we just need to rename the yml file in this branch.

cwsmith commented on .github/workflows/cmake.yml:
```diff
@@ -28,19 +29,27 @@ jobs:
+      run: |
+        sudo apt-get install doxygen
+        sudo apt-get install graphviz
+        sudo apt-get install libhdf5-mpi-dev
```
What requires `hdf5`?

PrestonH2O replied:
I don't remember, I think something to do with the serial compilation but it breaks without it.

cwsmith replied:
OK.  It would be nice to know which build breaks without these.  AFAIK, Cabana, Kokkos, Omega_h and MeshFields does not require these packages (doxygen, graphviz, and hdf5).

PrestonH2O replied:
Cabana fails without doxygen, checking the other 2 now ( I will edit this message as I find them )

cwsmith replied:
OK.  Thank you for checking/confirming.

cwsmith commented on .github/workflows/cmake.yml:
```diff
@@ -98,32 +107,63 @@ jobs:
+    - name: Convert Lcov to Cobertura
+      run: |
+        pip install lcov_cobertura
+        lcov_cobertura ${{runner.workspace}}/build-meshFields/coverage.info
```
We should pass the optional arguments to `lcov_cobertura`  (i.e., source directory, `--demangle` etc.) to see if the report changes.

PrestonH2O replied:
What changes are we looking to see in the report? The conversion currently does not change the report from the lcov version (which is an accurate report of code coverage)

cwsmith replied:
I don't know, but it should be pretty simple to add the args to `lcov_cobertura` and see what happens.

cwsmith replied:
OK.  Thank you for testing.

cwsmith reviewed (changes requested):
Looks good.  Thank you.  A few comments are below.

PrestonH2O commented:
I think I have addressed all comments and everything works as intended. Let me know if you want anything else before we go ahead and merge the pull request.
